src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1389,22 +1389,61 @@ private bool HasIdentityConversionInternal(TypeSymbol type1, TypeSymbol type2)
             return HasIdentityConversionInternal(type1, type2, IncludeNullability);
         }
 
+        /// <summary>
+        /// Returns true if:
+        /// - Either type has no nullability information (oblivious).
+        /// - Both types cannot have different nullability at the same time.
```

Comment:
>Both types cannot have different nullability at the same time. [](start = 14, length = 62)

Could we say: "Both types cannot have different nullability at the same time (distinguishing "nullable because of annotation" and "nullable because of constraint")"?

Reply:
> Could we say: "Both types cannot have different nullability at the same time (distinguishing "nullable because of annotation" and "nullable because of constraint")"?

I think there is no such thing as "nullable because of constraint", constrains cannot make things nullable, they can only make them not-nullable.

---
In reply to: [232011129](https://github.com/dotnet/roslyn/pull/30913#discussion_r232011129) [](ancestors = 232011129)

Reply:
I understand, but the point is the constraint is what determines if the type is nullable or not nullable.
Some other wording might work better: "nullable because unconstrained for nullability"?

Reply:
> Some other wording might work better: "nullable because unconstrained for nullability"?

How about "including the case of type parameters that by themselves can represent nullable and not nullable reference types"? 

---
In reply to: [232072293](https://github.com/dotnet/roslyn/pull/30913#discussion_r232072293) [](ancestors = 232072293)

Reply:
Sounds good